pkg/proxy/iptables/proxier.go

```diff
@@ -0,0 +1,567 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iptables
+
+/*
+NOTE: this needs to be tested in e2e since it uses iptables for everything.
+*/
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/proxy"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/util"
+	utilexec "k8s.io/kubernetes/pkg/util/exec"
+	utiliptables "k8s.io/kubernetes/pkg/util/iptables"
+	"k8s.io/kubernetes/pkg/util/slice"
+)
+
+// NOTE: It will be tricky to determine the exact version needed.
```

Comment:
Ok, I'm going to convert to using that vendored semver package in the next
commit then and fix the docs here.

On Mon, Aug 10, 2015 at 8:19 PM, Tim Hockin <notifications@github.com>
wrote:

> In pkg/proxy/iptables/proxier.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/9210#discussion_r36700321>
> :
>
> > +	"strconv"
> > +	"strings"
> > +	"sync"
> > +	"time"
> > +
> > +	"github.com/golang/glog"
> > +	"k8s.io/kubernetes/pkg/api"
> > +	"k8s.io/kubernetes/pkg/proxy"
> > +	"k8s.io/kubernetes/pkg/types"
> > +	"k8s.io/kubernetes/pkg/util"
> > +	utilexec "k8s.io/kubernetes/pkg/util/exec"
> > +	utiliptables "k8s.io/kubernetes/pkg/util/iptables"
> > +	"k8s.io/kubernetes/pkg/util/slice"
> > +)
> > +
> > +// NOTE: It will be tricky to determine the exact version needed.
>
> I think we can keep using version, I just want to be clear that this
> version is not sufficient on its own
> … <#14f1a207af2b06ed_>
> On Mon, Aug 10, 2015 at 3:55 PM, Benjamin Elder <notifications@github.com>
> wrote: In pkg/proxy/iptables/proxier.go <
> https://github.com/GoogleCloudPlatform/kubernetes/pull/9210#discussion_r36694602>
> : > + "strconv" > + "strings" > + "sync" > + "time" > + > + "
> github.com/golang/glog" > + "k8s.io/kubernetes/pkg/api" > + "
> k8s.io/kubernetes/pkg/proxy" > + "k8s.io/kubernetes/pkg/types" > + "
> k8s.io/kubernetes/pkg/util" > + utilexec "k8s.io/kubernetes/pkg/util/exec"
> > + utiliptables "k8s.io/kubernetes/pkg/util/iptables" > + "
> k8s.io/kubernetes/pkg/util/slice" > +) > + > +// NOTE: It will be tricky
> to determine the exact version needed. Previously the plan was to select
> between the old and new proxy depending on the host support for iptables.
> If we're no longer doing that, we can remove all the semver stuff. — Reply
> to this email directly or view it on GitHub <
> https://github.com/GoogleCloudPlatform/kubernetes/pull/9210/files#r36694602>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/9210/files#r36700321>
> .
>